src/toil/test/src/mpCopyTest.py

```diff
@@ -0,0 +1,79 @@
+import boto
+import uuid
+import math
+import itertools
+import sys
+import unittest
+import io
+from toil.mpcopy import copy_key_multipart
+
+
+chunk_size = int(2**20 * 5)  # if this value is changed the etags will not match
+test_upload_size = int(2**20 * 5.3)
+
+
+class CopyKeyMultipartTest(unittest.TestCase):
+    def setUp(self):
+        self.fileid = 'file_'+str(uuid.uuid1())
+        self.totalSize = test_upload_size
+        self.chunkSize = chunk_size
+        self.chunkCount = int(math.ceil(self.totalSize / float(self.chunkSize)))
+
+        self.s3 = boto.connect_s3()
+
+        self.src = {}
+        self.dst = {}
+
+        self.src['name'] = 'mp_copy_test_src_bucket_'+str(uuid.uuid1())
+        self.src['bucket'] = self.s3.create_bucket(self.src['name'])
+        self.src['url'] = "s3://%s/%s"%(self.src['name'], self.fileid)
+
+        self.dst['name'] = 'mp_copy_test_dst_bucket_'+str(uuid.uuid1())
+        self.dst['bucket'] = self.s3.create_bucket(self.dst['name'])
+        self.dst['url'] = "s3://%s/%s"%(self.dst['name'], self.fileid)
+
+        with open('/dev/urandom', 'r') as f:
+            upload = self.src['bucket'].initiate_multipart_upload(key_name=self.fileid)
+            start = 0
+            partindex = itertools.count()
+            try:
+                while start < self.totalSize:
+                    end = min(start + self.chunkSize, self.totalSize)
+                    b = io.BytesIO(f.read(self.chunkSize))
+
+                    upload.upload_part_from_file(fp=b,
+                                                 partindex=next(partindex) + 1,
+                                                 size=end - start)
+                    print "\ruploaded %d%%"%(int(math.ceil((float(start)* 100)/self.totalSize))),
+                    sys.stdout.flush()
+                    start = end
+                assert start == self.totalSize
+            except:
+                upload.cancel_upload()
+                raise
+            else:
+                upload.complete_upload()
+
+
+    def teardown(self):
+        for bkt in [self.src['bucket'], self.dst['bucket']]:
```

Comment:
Isn't it `tearDown()`, not `teardown()`?

Assuming I'm right, this method was never called. So it's disconcerting that the lack of clean up wasn't noticed earlier.